maxim/logger/components/generation.py

```
@@ -238,13 +256,20 @@ def add_message_(writer: LogWriter, id: str, message: GenerationRequestMessage):
             )
             return
+        messages, attachments = parse_attachments_from_messages([message])
+        if len(attachments) > 0:
+            for attachment in attachments:
+                Generation.add_attachment_(writer, id, attachment)
         BaseContainer._commit_(
-            writer, Entity.GENERATION, id, "update", {"messages": [message]}
+            writer, Entity.GENERATION, id, "update", {"messages": messages}
         )
 
     def add_message(self, message: GenerationRequestMessage):
```

Ruff (0.11.9)

Missing return type annotation for public function `add_message`

Add return type annotation: `None`

(ANN201)